invenio_communities/assets/semantic-ui/js/invenio_communities/members/members/MembersResultItem.js

```diff
@@ -0,0 +1,199 @@
+  result: PropTypes.object.isRequired,
+};
+
+class MemberViewResultItem extends Component {
+  updateVisibility = (value) => {
+    const { result } = this.props;
+    result.visibility = value;
+    // TODO: membersApi.edit(result); + NOTIFICATIONS
+  };
+
+  updateRole = (value) => {
+    const { result } = this.props;
+    result.role = value;
+    // TODO: membersApi.edit(result); + NOTIFICATIONS
+  };
+
+  render() {
+    const { result } = this.props;
+    //TODO: remove this check when backend implemented
+    const avatar = result.links.avatar
+      ? result.links.avatar
+      : "/static/images/square-placeholder.png";
+    return (
+      <Table.Row>
+        <Table.Cell>
+          <Grid textAlign="left" verticalAlign="middle">
+            <Grid.Column>
+              <Item className="flex-container" key={result.id}>
+                <Image
+                  src={avatar}
+                  avatar
+                  fallbackSrc="/static/images/square-placeholder.png"
+                />
+                <Item.Content className="ml-10">
+                  <Item.Header
+                    className={!result.member.description ? "mt-5" : ""}
+                  >
+                    <a
+                      className={result.member.is_group && "mt-10"}
+                      href={`/members/${result.id}`}
+                    >
+                      {result.member.name}
+                    </a>
+                    {result.member.is_group && (
+                      <Label className="ml-10">{i18next.t("Group")}</Label>
+                    )}
+                    {result.is_current_user && (
+                      <Label color="blue" className="ml-10">
+                        {i18next.t("You")}
+                      </Label>
+                    )}
+                  </Item.Header>
+                  {result.member.description && (
+                    <Item.Meta>
+                      <div
+                        className="truncate-lines-1"
+                        dangerouslySetInnerHTML={{
+                          __html: result.member.description,
+                        }}
+                      />
+                    </Item.Meta>
+                  )}
+                </Item.Content>
+              </Item>
+            </Grid.Column>
+          </Grid>
+        </Table.Cell>
+        <Table.Cell>{timestampToRelativeTime(result.created)}</Table.Cell>
+        <Table.Cell>
+          <MemberDropdown
+            initialValue={result.visibility}
+            options={dropdownOptionsGenerator(config.visibility.options)}
+            updateMember={this.updateVisibility}
+          />
+        </Table.Cell>
+        <Table.Cell>
+          <MemberDropdown
+            initialValue={result.role}
+            options={dropdownOptionsGenerator(config.role.options)}
+            updateMember={this.updateRole}
+          />
+        </Table.Cell>
+        <Table.Cell>
+          {result.is_current_user && (
```

Comment:
You might have to check here if the user is the only owner, then the user won't be able to leave... which triggers another corner case:
- I am the owner
- I set as owner another user because I want to leave
- I cannot leave because the search results are not refreshed after assigning another owner

Reply:
Yes, all the logic regarding the actions an user can do still needs to be defined and tested (in the UI), it will come in the next PR. Thanks for the corner case, will take it into consideration :+1: